### PR TITLE
feat: add ui pbc compatibility PLAT-1028

### PR DESCRIPTION
**Pull request by skawian**
[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

📓 &nbsp; **Related JIRA**
PLAT-1028

🤔 &nbsp; **What does this PR do?**

- Enables integration with genesis PBCs

🚀 &nbsp; **Where should the reviewer start?**

Add file / directory pointers.

📑 &nbsp; **How should this be tested?**

### reporting 

```
npx -y @genesislcap/genx@latest init prtest -x --ref ms/pbc-integration --no-shell && \
cd prtest && \
git init && git add --all && git commit -m 'init' && cd .. && \
npx -y @genesislcap/genx@latest add prtest -s pbc-reporting-seed && \
git status
```

### notify

```
npx -y @genesislcap/genx@latest init prtest2 -x --ref ms/pbc-integration --no-shell && \
cd prtest2 && \
git init && git add --all && git commit -m 'init' && cd .. && \
npx -y @genesislcap/genx@latest add prtest2 -s pbc-notify-seed -x && \
git status
```

Add pbc files to pbc folder and run the application

✅ &nbsp; **Checklist**

- [x] I have tested my changes.
- [ ] I have added tests for my changes.
- [x] I have updated the project documentation to reflect my changes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -51,6 +51,7 @@ gradle-app.setting
 
 # Cache of project
 .gradletasknamecache
+build-cache
```
Comment (cistov) [inline .gitignore:54]: `build-cache/` already there

Comment (skawian) [reply .gitignore]: removed

File: client/package.json [hunk 2]
```diff
@@ -106,5 +109,23 @@
     "@microsoft/fast-web-utilities": "^5.1.0",
     "rxjs": "^7.5.4",
     "tslib": "^2.3.1"
-  }
+  },
+  "overrides": {
```
Comment (MrBrunoWolff) [inline client/package.json:113]: question:

why do we need the overrides?

**Review by MrBrunoWolff — COMMENTED** [review]
NICE! lots of cleanups/updates!

Leaving one question only + need to test the seed now.

**Comment by cistov** [comment]
```
{
      navId: 'side',
      title: 'Special Report',
      routePath: 'reporting/foo', // < example if there were child routes
      icon: {
        name: 'file-csv',
        variant: 'solid',
      },
    },
```

Special report from reporting doesn't get added - could be just a bad nav id `side`. In foundation shell `export type SideNavTargetId = 'side-nav-start' | 'side-nav' | 'side-nav-end';`

**Comment by cistov** [comment]
With Notify, it's failing on `import { FoundationAlerts } from '@genesislcap/pbc-notify-ui'` import:

```
registrations.ts:1 Uncaught (in promise) SyntaxError: The requested module '/node_modules/.vite/deps/@genesislcap_pbc-notify-ui.js?v=2bc53113' does not provide an export named 'FoundationAlerts' (at registrations.ts:1:10)
```

https://github.com/genesislcap/genesis-notify/blob/master/genx/client/src/pbc/notify/registrations.ts

**Comment by cistov** [comment]
Screenshot 2024-03-11 at 14 21 55

Changing navId to header makes the Special Report appear. 

`routePath: 'reporting/foo', // < example if there were child routes` this doesn't seem to set a child route unless I'm missing something

**Comment by cistov** [comment]
Looks like config issues in both PBCs, esp. notify.

Child route config doesn't seem to work if I understand it correctly, which would be foundation-shell issue.

**Review by cistov — APPROVED** [review]

**Pull request merged**